ScorchApp/src/components/util/Searcher.vue

```diff
@@ -0,0 +1,200 @@
+                    </a>
+                </li>
+                <li class="page-item clickable" v-for="(page, index) in pagesToShow" :key="index" v-on:click="currentPage = page" v-bind:class ="{active : page == currentPage}">
+                    <a class="page-link" >{{page}}</a>
+                </li>
+                
+                <li class="page-item clickable" v-on:click="currentPage = Math.min(currentPage+1, numPages)">
+                    <a class="page-link" >
+                        <span aria-hidden="true">&raquo;</span>
+                    </a>
+                </li>
+            </ul>
+        </div>
+    </div>
+    
+</template>
+
+<script>
+    import filter from 'lodash/filter'
+    import orderBy from 'lodash/orderBy'
+    import sortBy from 'lodash/sortBy'
+
+export default {
+    name : 'searcher',
+    
+    data() {
+      return {
+        searchTerm : '',
+        sortKey : '',
+        sortDirection : 'asc',
+        objectKeys: [],
+        keysPopulated : false,
+        currentPage : 1
+      }
+    },
+    props: ['searchData', 'limitPerPage'],
+    methods: {
+        selectRow(row){
+            this.$emit('search-row-selected', row);
+        },
+        sortByForm(key){
+            if(this.sortKey === key){
+                this.sortDirection = this.sortDirection === 'desc' ? 'asc' : 'desc';
+            }
+            else{
+                this.sortKey = key;
+                this.sortDirection = 'asc';
+            }
+        },
+        populateObjectKeys(results){
+            this.keysPopulated = true;
+            var allKeys = [];
+            results.forEach(result => {
+                Object.keys(result).forEach(key => {
+                    if(!allKeys.includes(key))
+                        allKeys.push(key);
+                });
+            });
+            if(allKeys.length > 0)
+            {
+                this.sortKey = allKeys[0];
+            }
+            this.objectKeys = allKeys;
+        },
+        includesAllChars(queryString, element){
+            var result = false;
+            var index = 0;
+            var qsChars = queryString.toUpperCase().split('');
+            var elemChars = element.toUpperCase().split('');
+            elemChars.forEach(elemChar => {
+                if(qsChars[index] === elemChar){
+                    index++;
+                }
+                if(qsChars.length == index){
+                    result = true;
+                }
+            });
+
+            return result;
+        }
+    },
+    computed: {
+        searchResults : function(){
+            let results = this.searchData;
+            if(!this.keysPopulated && this.searchData.length > 0)
+                this.populateObjectKeys(results);
+
+            results = orderBy(results, [this.sortKey], [this.sortDirection]);
+
+            results = filter(results, row => {
+                var display = this.searchTerm ? false : true;
+                for(var propKey in row){
```

Comment:
I don't think you need to check if `row[propKey] == this.searchTerm` as that is covered in your includeAllChars function . Because if the includeAllChars returns false, then we know already row[propKey] can't equal this.searchTerm. 

I think you should modify the includeAllChars to also ignore types. (If a name is like Ar15, and I type in 15, I should get that and also things that are AC 15 and weigh 15)